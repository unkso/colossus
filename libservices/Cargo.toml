[package]
name = "libservices"
version = "0.1.0"
edition = "2018"
description = "Types and macros for Rocket services"
license = "MIT"
repository = "https://github.com/Unkso/Colossus"
keywords = ["web", "rocket", "api", "app"]
categories = ["web-programming"]
readme = "README.md"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
thiserror = "1.0"
async-trait = "0.1"
rocket = { git = "https://github.com/SergioBenitez/Rocket", branch = "master" }
libservices_codegen = { path = "../libservices_codegen" }
