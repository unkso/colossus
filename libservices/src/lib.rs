mod errors;
mod types;

pub use errors::ServiceError;
pub use libservices_codegen::AttachableService;
pub use types::{AbstractService, AttachableService, Initializer};
