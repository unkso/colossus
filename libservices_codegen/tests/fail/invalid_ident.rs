#![allow(dead_code)]
use libservices::{AbstractService, AttachableService, ServiceError};

#[derive(AttachableService)]
#[service(alias = 42)]
struct Service;

impl Service {
    pub fn new() -> Self {
        Self
    }
}

#[rocket::async_trait]
impl AbstractService for Service {
    type Error = ServiceError;

    async fn init(_rocket: &rocket::Rocket<rocket::Build>) -> Result<Self, Self::Error> {
        Ok(Self::new())
    }
}

fn main() {}
