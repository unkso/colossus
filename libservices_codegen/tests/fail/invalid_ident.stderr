error: value in `#[service(alias = "value")]` is not a valid ident
 --> $DIR/invalid_ident.rs:5:1
  |
5 | #[service(alias = 42)]
  | ^
